solver/src/solver/paraswap_solver/api.rs

```diff
@@ -29,45 +29,70 @@ pub struct DefaultParaswapApi {
 
 #[async_trait::async_trait]
 impl ParaswapApi for DefaultParaswapApi {
-    async fn price(&self, query: PriceQuery) -> Result<PriceResponse> {
+    async fn price(&self, query: PriceQuery) -> Result<PriceResponse, ParaswapResponseError> {
+        let query_str = format!("{:?}", &query);
         let url = query.into_url();
         tracing::debug!("Querying Paraswap API (price) for url {}", url);
-        let text = reqwest::get(url)
+        let response_text = reqwest::get(url)
             .await
-            .context("PriceQuery failed")?
+            .map_err(ParaswapResponseError::Send)?
             .text()
-            .await?;
-        tracing::debug!("Response from Paraswap API (price): {}", text);
-
-        serde_json::from_str::<PriceResponse>(&text)
-            .context(format!("PriceQuery result parsing failed: {}", text))
+            .await
+            .map_err(ParaswapResponseError::TextFetch)?;
+        tracing::debug!("Response from Paraswap API (price): {}", response_text);
+        let raw_response = serde_json::from_str::<RawPriceResponse>(&response_text);
```

Comment:
ubernit: You can reduce the number of branches you need to handle :point_down: by mapping the error and using the `?` try operator first.
```suggestion
        let raw_response = serde_json::from_str::<RawPriceResponse>(&response_text)
            .map_err(ParaswapResponseError::DeserializeError)?;
```

Just thought it would be more consistent with the error handling in the rest of the function, but no strong feelings about this one (since there is barely any "rightward drift" here).